kube-client/src/lib.rs

```diff
@@ -122,3 +122,333 @@ cfg_error! {
+            match status {
+                WatchEvent::Modified(o) => {
+                    let s = o.status.as_ref().expect("status exists on pod");
+                    let phase = s.phase.clone().unwrap_or_default();
+                    if phase == "Running" {
+                        break;
+                    }
+                }
+                WatchEvent::Error(e) => assert!(false, "watch error: {}", e),
+                _ => {}
+            }
+        }
+
+        // Verify exec works and we can get the output
+        {
+            let mut attached = pods
+                .exec(
+                    "busybox-kube2",
+                    vec!["sh", "-c", "for i in $(seq 1 3); do echo $i; done"],
+                    &AttachParams::default().stderr(false),
+                )
+                .await?;
+            let stdout = tokio_util::io::ReaderStream::new(attached.stdout().unwrap());
+            let out = stdout
+                .filter_map(|r| async { r.ok().and_then(|v| String::from_utf8(v.to_vec()).ok()) })
+                .collect::<Vec<_>>()
+                .await
+                .join("");
+            attached.await;
+            assert_eq!(out.lines().count(), 3);
+            assert_eq!(out, "1\n2\n3\n");
+        }
+
+        // Verify we can write to Stdin
+        {
+            use tokio::io::AsyncWriteExt;
+            let mut attached = pods
+                .exec(
+                    "busybox-kube2",
+                    vec!["sh"],
+                    &AttachParams::default().stdin(true).stderr(false),
+                )
+                .await?;
+            let mut stdin_writer = attached.stdin().unwrap();
+            let mut stdout_stream = tokio_util::io::ReaderStream::new(attached.stdout().unwrap());
+            let next_stdout = stdout_stream.next();
+            stdin_writer.write(b"echo test string 1\n").await?;
+            let stdout = String::from_utf8(next_stdout.await.unwrap().unwrap().to_vec()).unwrap();
+            println!("{}", stdout);
+            assert_eq!(stdout, "test string 1\n");
+
+            // AttachedProcess resolves with status object.
+            // Send `exit 1` to get a failure status.
+            stdin_writer.write(b"exit 1\n").await?;
+            if let Some(status) = attached.await {
+                println!("{:?}", status);
+                assert_eq!(status.status, Some("Failure".to_owned()));
+                assert_eq!(status.reason, Some("NonZeroExitCode".to_owned()));
+            }
+        }
+
+        // evict the pod
+        let ep = EvictParams {
+            delete_options: Some(DeleteParams {
+                grace_period_seconds: Some(0),
+                ..DeleteParams::default()
+            }),
+            ..EvictParams::default()
+        };
+        let eres = pods.evict("busybox-kube2", &ep).await?;
+        assert_eq!(eres.code, 201); // created
+        assert_eq!(eres.status, "Success");
```

Comment:
I think you meant to add this to the next test (`can_get_pod_logs_and_evict_pod`).